nixos/modules/services/torrent/qbittorrent.nix

```diff
@@ -0,0 +1,200 @@
+{
+  config,
+  pkgs,
+  lib,
+  utils,
+  ...
+}:
+let
+  cfg = config.services.qbittorrent;
+  inherit (builtins) concatStringsSep isAttrs isString;
+  inherit (lib)
+    literalExpression
+    getExe
+    mkEnableOption
+    mkOption
+    mkPackageOption
+    mkIf
+    maintainers
+    escape
+    collect
+    mapAttrsRecursive
+    ;
+  inherit (lib.types)
+    str
+    port
+    path
+    nullOr
+    unspecified
+    ;
+  inherit (lib.generators) toINI mkKeyValueDefault mkValueStringDefault;
+  gendeepINI = toINI {
+    mkKeyValue =
+      let
+        sep = "=";
+      in
+        k: v:
+        if isAttrs v then
+          concatStringsSep "\n"
+            (
+              collect
+                isString
+                (mapAttrsRecursive (path: value: "${escape [ sep ] (concatStringsSep "\\" ([ k ] ++ path))}${sep}${mkValueStringDefault {} value}") v)
+            )
+        else
+          mkKeyValueDefault { } sep k v;
+  };
+in
+{
+  options.services.qbittorrent = {
+    enable = mkEnableOption "qbittorrent, BitTorrent client.";
+
+    package = mkPackageOption pkgs "qbittorrent-nox" { };
+
+    user = mkOption {
+      type = str;
+      default = "qbittorrent";
+      description = "User account under which qbittorrent runs.";
+    };
+
+    group = mkOption {
+      type = str;
+      default = "qbittorrent";
+      description = "Group under which qbittorrent runs.";
+    };
+
+    profileDir = mkOption {
+      type = path;
+      default = "/var/lib/qBittorrent/";
+      description = "the path passed to qbittorrent via --profile.";
+    };
+
+    openFirewall = mkOption {
+      default = false;
```

Comment:
seems to be missing a `type` for some reason? Or we could just make this `mkEnableOption` and save some boilerplate.

Reply:
that reason would be a lack of thinking on my behalf  :)))